src/config.rs

```diff
@@ -418,6 +418,10 @@ impl<'a> Cfg<'a> {
     }
 
     pub(crate) fn get_self_update_mode(&self) -> Result<SelfUpdateMode> {
+        if self.process.var("CI").is_ok() {
+            return Ok(SelfUpdateMode::Disable);
```

Comment:
Alternatively, we could return `CheckOnly`?

Reply:
@djc I guess disabled would still be better. It seems to me that the whole auto self update thing in the CI context is only introducing overheads.

Reply:
That's what I thought as well. 👍